doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/event-listeners-development-con.adoc

```diff
@@ -0,0 +1,14 @@
+[id='event-listeners-development-con_{context}']
+= Practices for development of event listeners
+
+The {PROCESS_ENGINE} calls event listeners during processing of events or tasks. The calls happen within {PROCESS_ENGINE} transactions are committed and block execution. Therefore, the event listener can affect the logic and performance of the {PROCESS_ENGINE}.
+
+To ensure minimal disruption, observe the following practices.
+
+* Any action must be as short as possible.
+* A listener class must not have a state. The {PROCESS_ENGINE} can destroy and re-create a listener class at any time.
+* If the listener modifies any object that exists outside the scope of the listener method and is not already a part of the transaction, the listener must also enlist the object into the transaction. The transaction might be rolled back, and in this case, if the modified object is not a part of the transaction, the state of the object becomes inconsistent. *EXAMPLE CODE FOR RETRIEVING THE TRANSACTION FROM THE EVENT AND ENLISTING A TASK INTO THE TRANSACTION REQUIRED*
```

Comment:
don't talk about objects but resources. This is more meanful when talking about transactions as it talks about a jms queue or an in memory object transaction.

Reply:
This is important, thanks, but I do also need a very simple code sample. Right now I could not even work out how to retrieve the transaction object.